R/utilities-figures.R

```diff
@@ -699,11 +703,18 @@ createPlotsFromExcel <- function(
 #'
 #' @keywords internal
 #' @noRd
-.calculateLimits <- function(x) {
-  limits <- c(
-    (if (min(x, na.rm = TRUE) <= 0) 1.01 else 0.99) * min(x, na.rm = TRUE),
-    (if (max(x, na.rm = TRUE) > 0) 1.01 else 0.99) * max(x, na.rm = TRUE)
-  )
+.calculateLimits <- function(x, scaling = NULL) {
+  if (!is.null(scaling) && scaling == "log") {
+    limits <- c(
+      min(x[x > 0], na.rm = TRUE) * 0.9,
+      max(x[x > 0], na.rm = TRUE) * 1.1
+    )
+  } else {
+    limits <- c(
+      (if (min(x, na.rm = TRUE) <= 0) 1.01 else 0.99) * min(x, na.rm = TRUE),
+      (if (max(x, na.rm = TRUE) > 0) 1.01 else 0.99) * max(x, na.rm = TRUE)
+    )
+  }
```

Comment:
Is this function used to add "blank" area below and above data points ?
If yes, I would suggest to use [the expand argument](https://ggplot2.tidyverse.org/reference/expansion.html)
instead of the limit one.

Reply:
The function is used to pre-calculate limits, which are necessary for assembling plots into facets. Limits also help with filtering data for log-transformed axis plotting to avoid the ggplot log transform warning message. Tried `scales::expand_range`, but it did not produce the desired result.